site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+* [Headless Services](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) with selectors. Headless services without selectors SHOULD NOT be supported due to the possibility of matching `Services` with `type: ExternalName`.
+* `StatefulSet` resources
+
+#### `Service` without selectors
+
+An alternate pattern additionally supported by this approach would be to target a `Service` without selectors as the `parentRef`. This could be a clean way to create a pure routing construct and abstract a logical frontend, as traffic would resolve to a `backendRef` `Service` with selectors defined on the `HTTPRoute`, or receive a 4xx/5xx error response if no matching path or valid backend was found.
+
+#### StatefulSet
+
+[`StatefulSet`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) resources MAY be supported as a `backendRef`, but specifics of this integration have not yet been clarified in the full Gateway API spec.
+
+#### Multicluster support with `ServiceImport`
+
+`ServiceImport` resources allocate a virtual IP in the cluster, so MAY be allowed as a `parentRef`.
+
+`ServiceImport` would remain a valid backend option for `HTTPRoute` resources (but _not_ currently a requirement for core conformance), and could be specified alongside a `Service` `backendRef` to split traffic across clusters within a `ClusterSet` (as defined in the [Multi-cluster Service (MCS) APIs project](https://github.com/kubernetes-sigs/mcs-api)). This could be a way to solve the need described in [A use case for using Gateway APIs in Multi-Cluster](https://docs.google.com/document/d/1bjr0uAVMmEtTX4mpU_aGXXapQFsEz_Qt0OnDoVswEEI/edit).
+
+### `hostnames` field
+
+GAMMA implementations SHOULD NOT infer any functionality from the `hostnames` field on `HTTPRoute` due to current under-specification and reserved potential for future usage or API changes. For the use case of filtering incoming traffic from selected hostnames, it is recommended to guide users toward configuring [`HTTPHeaderMatch`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPHeaderMatch) rules for the [`Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) header. Functionality to be explored in future GEPs may include supporting concurrent usage of an `HTTPRoute` traffic configuration for multiple North/South `Gateways` and East/West mesh use cases or redirection of egress traffic to an in-cluster `Service`.
+
+### Limiting graph depth, preventing conflicts and cycles
+
+A `Service` may only be specified as a `parentRef` of an `HTTPRoute` if it is:
+
+* NOT already specified as a `parentRef` of another `HTTPRoute` [in the same namespace]
+    * This requirement may be relaxed later to follow the existing [precedence rules](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule), but is intended to simplify initial implementation.
```

Comment:
It seems like we may need some kind of conflict resolution logic here similar to what we already have in place for Gateway API.

Reply:
@robscott How does this look?
```suggestion
    * Resolving conflicts from multiple `HTTPRoutes` attempting to target the same `Service` as a `parentRef` should follow the existing [conflict guidelines](https://gateway-api.sigs.k8s.io/concepts/guidelines/#conflicts).
      * When an existing `HTTPRoute` targeting the `Service` as a `parentRef` has already been accepted, any `HTTPRoute` with a newer creation timestamp should be rejected, because the match target considered for the specificity rules (the `Service`) is identical.
      * If creation timestamps are identical (from applying both routes simultaneously), only the `HTTPRoute` with a `name` appearing first in alphabetical order should be accepted.
      * Any rejected `HTTPRoute` should set an `Accepted` condition with `status: "False"`.
    * This requirement may be relaxed later to follow the existing [precedence rules for merging configuration](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule), but is intended to simplify initial implementation.
```

Reply:
LGTM but I plan to submit a close followup to allow the HTTP merging rules